tests/check/check.go

```diff
@@ -125,63 +133,88 @@ func runTests(t *testing.T, schemaVersion string, client ClientInterface) {
+	name := test.Name
+	if contextTupleTest {
+		name += "_ctxTuples"
+	}
 
-			storeID := resp.GetId()
+	if contextTupleTest && len(test.Stages) > 1 {
+		// we don't want to run special contextual tuples test for these cases
+		// as multi-stages test has expectation tuples are in system
+		return
+	}
 
-			for _, stage := range test.Stages {
+	t.Run(name, func(t *testing.T) {
+		t.Parallel()
+		ctx := context.Background()
 
-				var typedefs []*openfgapb.TypeDefinition
-				if schemaVersion == typesystem.SchemaVersion1_1 {
-					typedefs = parser.MustParse(stage.Model)
-				} else {
-					typedefs = v1parser.MustParse(stage.Model)
-				}
+		resp, err := client.CreateStore(ctx, &openfgapb.CreateStoreRequest{Name: name})
+		require.NoError(t, err)
 
-				_, err = client.WriteAuthorizationModel(ctx, &openfgapb.WriteAuthorizationModelRequest{
-					StoreId:         storeID,
-					SchemaVersion:   schemaVersion,
-					TypeDefinitions: typedefs,
+		storeID := resp.GetId()
+
+		for _, stage := range test.Stages {
+
+			var typedefs []*openfgapb.TypeDefinition
+			if schemaVersion == typesystem.SchemaVersion1_1 {
+				typedefs = parser.MustParse(stage.Model)
+			} else {
+				typedefs = v1parser.MustParse(stage.Model)
+			}
+
+			_, err = client.WriteAuthorizationModel(ctx, &openfgapb.WriteAuthorizationModelRequest{
+				StoreId:         storeID,
+				SchemaVersion:   schemaVersion,
+				TypeDefinitions: typedefs,
+			})
+			require.NoError(t, err)
+
+			if len(stage.Tuples) > 0 && !contextTupleTest {
+				_, err = client.Write(ctx, &openfgapb.WriteRequest{
+					StoreId: storeID,
+					Writes:  &openfgapb.TupleKeys{TupleKeys: stage.Tuples},
 				})
 				require.NoError(t, err)
 
-				if len(stage.Tuples) > 0 {
-					_, err = client.Write(ctx, &openfgapb.WriteRequest{
-						StoreId: storeID,
-						Writes:  &openfgapb.TupleKeys{TupleKeys: stage.Tuples},
-					})
-					require.NoError(t, err)
+			}
+
+			for _, assertion := range stage.CheckAssertions {
+				ctxTuples := assertion.ContextualTuples
+				if contextTupleTest {
+					ctxTuples = append(ctxTuples, stage.Tuples...)
 				}
 
-				for _, assertion := range stage.CheckAssertions {
-					resp, err := client.Check(ctx, &openfgapb.CheckRequest{
-						StoreId:  storeID,
-						TupleKey: assertion.Tuple,
-						ContextualTuples: &openfgapb.ContextualTupleKeys{
-							TupleKeys: assertion.ContextualTuples,
-						},
-						Trace: true,
-					})
-
-					if assertion.ErrorCode == 0 {
-						require.NoError(t, err)
-						require.Equal(t, assertion.Expectation, resp.Allowed, assertion)
-					} else {
-						require.Error(t, err)
-						e, ok := status.FromError(err)
-						require.True(t, ok)
-						require.Equal(t, assertion.ErrorCode, int(e.Code()))
-					}
+				resp, err := client.Check(ctx, &openfgapb.CheckRequest{
+					StoreId:  storeID,
+					TupleKey: assertion.Tuple,
+					ContextualTuples: &openfgapb.ContextualTupleKeys{
+						TupleKeys: ctxTuples,
+					},
+					Trace: true,
+				})
```

Comment:
if i'm understanding correctly, the test setup may include contextual tuples already? 

if so, would it make sense to, in the YAML files, merge both tuples and contextual tuples into only tuples, and then here we do

```golang
if contextTupleTest {
     ctxTuples = stage.Tuples
}

resp, err := client.Check(ctx, &openfgapb.CheckRequest{
     StoreId:  storeID,
     TupleKey: assertion.Tuple,
     ContextualTuples: &openfgapb.ContextualTupleKeys{
          TupleKeys: ctxTuples,
	},
     Trace: true,
})
```

Reply:
For most case, yes. However, for multi-stage test, we will not run the context test.

Reply:
Also, there are error cases which we want to target contextual tuples only